src/lms-ds-loader/edfi_lms_ds_loader/sql_adapter.py

```diff
@@ -5,17 +5,18 @@
 
 # Developer note: this adapter module is deliberately not unit tested.
 
-from typing import Any, Callable, List, Optional
+from typing import Any, Callable, List, TypeVar
 
 from sqlalchemy.engine.base import Engine as sa_Engine
-from sqlalchemy.engine.result import ResultProxy as sa_Result
 from sqlalchemy.orm import sessionmaker, Session as sa_Session
 from sqlalchemy.exc import ProgrammingError
 
+T = TypeVar('T')
+
 
 def execute_transaction(
-    engine: sa_Engine, function: Callable[[sa_Session], Optional[sa_Result]]
-) -> Optional[sa_Result]:
+    engine: sa_Engine, function: Callable[[sa_Session], T]
+) -> T:
```

Comment:
I figured out how to use generics here! That fixed some confusing type issues I  had before.

Reply:
Very cool!